app/src/test/java/ar/scacchipa/jsonplaceholderviewer/PostRespositoryUnitTest.kt

```diff
@@ -0,0 +1,84 @@
+package ar.scacchipa.jsonplaceholderviewer
+
+import ar.scacchipa.jsonplaceholderviewer.data.*
+import kotlinx.coroutines.runBlocking
+import org.junit.After
+import org.junit.Test
+import org.koin.core.context.startKoin
+import org.koin.core.context.stopKoin
+import org.koin.dsl.module
+import org.koin.test.KoinTest
+import org.koin.test.inject
+import retrofit2.Response
+import kotlin.test.assertEquals
+import kotlin.test.assertTrue
+
+
+class MockSourceData: ITypicodeSourceData {
+    override suspend fun getPosts(): Response<List<Post>> {
+        return Response.success( getMockPosts() )
+    }
+
+    override suspend fun getComment(id: String): Response<List<Comment>> {
+        return if (id.toInt() == 5) {
+            Response.success( getMockComments() )
+        } else {
+            Response.success( listOf() )
+        }
+    }
+}
+
+class PostRespositoryUnitTest: KoinTest {
+
+    private val postRepository: IPostRepository by inject()
+
+    @Test
+    fun shouldGetNinePost() = runBlocking {
+        startKoin {
```

Comment:
sorry for my lack of knoledge on koin , maybe this is the best way , but maybe we can found a way to define all the depency graph in just one place for the testing, so we dont need to startKoin() and define specific graph sections for each test. Seems like there should be a better way.

Reply:
startKoin was move out of test function